File: 228. Summary Ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if len(nums) == 0:
            return []
        if len(nums) == 1:
            return [str(nums[0])]
        arrow = '->'
        inrange = False
        rangestart = None
        result = []
        for i, num in enumerate(nums):
            if inrange:
                if i == len(nums)-1:
                    result.append(str(rangestart) + arrow + str(num))
                else:
                    if nums[i+1] == num + 1:
                        inrange = True
                    else:
                        inrange = False
                        result.append(str(rangestart)+arrow+str(num))
            else:
                if i == len(nums)-1:
                    result.append(str(num))
                else:
                    if nums[i+1] == num + 1:
                        inrange = True
                        rangestart = num
                    else:
                        result.append(str(num))
        return result

File: 228. Summary Ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> ans = new ArrayList<>();
        if (nums==null || nums.length==0) {
            return ans;
        } else if (nums.length==1) {
            ans.add(String.valueOf(nums[0]));
            return ans;
        }
        int start=nums[0], end = nums[0];
        for (int i=1; i<nums.length; i++) {
            if (nums[i] - end == 1) {
                end = nums[i];
            } else {
                if (end==start) {
                    ans.add(String.valueOf(start)); 
                } else {
                    ans.add(String.valueOf(start) + "->" + String.valueOf(end));
                }
                start = nums[i];
                end = nums[i];
            }
        }
        if (end==start) {
                    ans.add(String.valueOf(start));
        } else {
            ans.add(String.valueOf(start) + "->" + String.valueOf(end));
        }
        return ans;
    }
}